src/main/java/org/broadinstitute/hellbender/engine/ReadWalker.java

```diff
@@ -88,6 +87,14 @@ public void traverse() {
         logger.info(countedFilter.getSummaryLine());
     }
 
+    /**
+     * Returns an interval for the read.
+     * Note: some walkers must be able to work on any read, including those whose coordinates do not form a valid SimpleInterval.
+     * So here we check this condition and create empty intervals for those bad reads.
```

Comment:
@akiezun No need to revisit `SimpleInterval` semantics -- the approach in this branch of avoiding creating intervals from these reads in the engine is perfectly fine, and consistent with what we were already doing for unmapped reads.

I'll also note that we already had an alpha-1 ticket to fix this problem (https://github.com/broadinstitute/gatk/issues/373, which this PR should close), and the proposed fix was exactly the one you settled on here.